java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java

```diff
@@ -856,18 +850,45 @@ private static void alignErrata(Channel src, Channel tgt, Collection<ErrataFilte
         ErrataManager.mergeErrataToChannel(user, includedErrata, tgt, src, false, false);
     }
 
+    private static Map<Boolean, List<Errata>> filterErrata(Channel src, Collection<ErrataFilter> errataFilters) {
+        Set<ErrataFilter> errataFilterSet = new HashSet<>(errataFilters);
+        Predicate<Errata> denyPredicate = errataFilterSet.stream()
+                .filter(f -> f.getRule() == ContentFilter.Rule.DENY)
+                .map(f -> (Predicate) f)
+                .reduce(x -> false, (f1, f2) -> f1.or(f2));
```

Comment:
I do not really understand all of this yet, but should "deny" filters not use `and` condition while only "allow" filters should use `or` ?
Or is this logic implemented at a different place?

Reply:
In the newest patches it was refactored so that there is only one pass when filtering, but the `or` made sense here. The logic when filtering, say, packages is as follows:
- go through all the packages
- if a package satisfies any DENY filter (that's why the `or` in the predicate), it's filtered out
- if the package was filtered out by DENY filter, but it passes any ALLOW filter (again, the `or` here), it's back again.

In the current state of the PR, I re-wrote it, so that it 1-pass - see the javadoc of [this method](https://github.com/uyuni-project/uyuni/blob/1a06e4d61a81f81ac013d5c68134e2332d12afdd/java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java#L866).

Reply:
Makes sense.